ocaml/xenopsd/lib/xenops_sandbox.ml

```diff
@@ -104,59 +104,104 @@ end = struct
     )
 end
 
-module Varstore_guard = struct
-  let daemon = "varstored"
+module type Chrooter = sig
+  val prepare : domid:int -> vm_uuid:string -> Chroot.Path.t -> string
+  (** [prepare ~domid path] creates an empty [path] file owned by [domid] inside
+      the chroot for [domid] and returns the absolute path to it outside the
+      chroot *)
 
-  let varstored_chroot ~domid ~vm_uuid = Chroot.of_domid ~daemon ~domid ~vm_uuid
+  val start :
+       string
+    -> vm_uuid:string
+    -> domid:int
+    -> paths:Chroot.Path.t list
+    -> Chroot.t * string
+  (** [start dbg ~vm_uuid ~domid ~paths] prepares a chroot for [domid], and asks
+      the guard to create a socket restricted to [vm_uuid]. Also creates
+      empty files specified in [path] owned by [domid] user. *)
+
+  val read : domid:int -> Chroot.Path.t -> vm_uuid:string -> string
+
+  val stop : string -> domid:int -> vm_uuid:string -> unit
+  (** [strop dbg ~domid ~vm_uuid] *)
+end
+
+module type Guard = sig
+  val daemon_name : string
+
+  val create : string -> vm_uuid:Uuidm.t -> domid:int -> path:string -> unit
```

Comment:
Based on the type, `vm` would be good enough as a label. But this might be less convenient on the call side where `vm` could be used for something else.